### ISSUE TITLE
How to transform directed graph to undirected graph?

### ISSUE THREAD

**Issue by GeorgeScrivener** [petgraph/petgraph#291]
I have a directed graph and need to view it as an undirected graph.  Is this implemented already or do I need to create my own adaptor?

**Comment by jmcomets** [comment]
Could you provide some details on what you mean by "viewing it as an undirected graph"?

If you're speaking of an adapter for traversals & other algorithms, then no it isn't implemented.

**Comment by GeorgeScrivener** [comment]
Yeah I meant an adaptor so I can pass my directed graph in to algorithms and have them treat it as an undirected graph.
I'm happy to make a PR for this if you think it would be useful?  I think that copying `Reversed` would be a good start.

**Comment by jmcomets** [comment]
I'd definitely welcome that! Note that there's quite a few traits that you need to implement for most algorithms to work. I agree that copying `Reversed` is a good start. 👍

**Comment by timthelion** [comment]
FYI you can do this very simply right now with an unsafe transmute: https://timothy.hobbs.cz/rust-play/petgraph-internals.html#The-Graph-data-structure

```rust
let undirected_tree : Graph<&str, &str, petgraph::Undirected> = unsafe {std::mem::transmute(tree.clone())};
draw_graph(&undirected_tree);
```

**Comment by GeorgeScrivener** [comment]
I'm having a look at this now and I'm unsure whether I should implement `IntoNeighborsDirected` for the `UndirectedAdaptor` (proposed name)?
Intuitively I'd say that it shouldn't be implemented as it is always undirected, however it is implemented for `Graph` whether or not the edges are directed.
@jmcomets what's your view on this?

**Comment by GeorgeScrivener** [comment]
PR for this here: #318
